lib/plottr_components/src/components/FirebaseLogin.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react'
+import { PropTypes } from 'prop-types'
+
+const FirebaseLoginConnector = (connector) => {
+  const {
+    platform: {
+      useSettingsInfo,
+      firebase: { startUI, firebaseUI, onSessionChange, fetchFiles },
+    },
+  } = connector
+
+  const FirebaseLogin = ({ setUserId, setFileList, receiveUserId }) => {
+    const [_settings, _size, saveSetting] = useSettingsInfo()
+    const [firebaseLoginComponentRef, setFirebaseLoginComponentRef] = useState(null)
+
+    useEffect(() => {
+      if (firebaseLoginComponentRef) {
+        const ui = firebaseUI()
+        startUI(ui, '#firebase_login_root')
+      }
+    }, [firebaseLoginComponentRef])
+
+    useEffect(() => {
+      onSessionChange((user) => {
```

Comment:
`onSessionChange` returns a function that you can call to deregister the listener.
(I need to update the README.)